src/ci/docker/disabled/dist-armv7r-none-eabihf/Dockerfile

```diff
@@ -0,0 +1,34 @@
+FROM ubuntu:16.04
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+  g++ \
+  make \
+  file \
+  curl \
+  ca-certificates \
+  python2.7 \
+  git \
+  cmake \
+  sudo \
+  xz-utils \
+  bzip2 \
+  libssl-dev \
+  pkg-config
+
+
+COPY scripts/sccache.sh /scripts/
+RUN sh /scripts/sccache.sh
+
+# GNU Arm Embedded Toolchain 7-2018-q2-update (June 27,2018)
+ENV BASE_URL=https://developer.arm.com/-/media/Files/downloads/gnu-rm/
+RUN curl -L $BASE_URL/7-2018q2/gcc-arm-none-eabi-7-2018-q2-update-linux.tar.bz2 | tar -xj
```

Comment:
Any reason we should use this specific GCC version rather than the one packaged by Ubuntu? Are there know codegen bugs?

Because if the Ubuntu version is enough we can build this target in the dist-various-1 image where all the Cortex-M targets are built.

Reply:
Another reason to avoid these toolchains is that these links are not permanent. For example the other docker image you added for the armebv7r target no longer builds because [this URL](https://github.com/rust-lang/rust/blob/master/src/ci/docker/disabled/dist-armebv7r-none-eabihf/Dockerfile#L22) no longer exists.

Reply:
> Any reason we should use this specific GCC version
Do you mean the default Ubuntu 16.04 package or the one in the PPA channel (ppa:team-gcc-arm-embedded/ppa)?

I use 7-2018-q2-update mainly for Armv8r, BTW I can use Ubuntu package here.

Reply:
> Another reason to avoid these toolchains is that these links are not permanent. For example the other docker image you added for the armebv7r target no longer builds because this URL no longer exists.

Here is "mandatory" because it is an armeb toolchain. I'll update the link. The issue here is that use 'latest' folder instead of '7.2.1' and latest now is 7.3.x.

Reply:
Why is it mandatory? The `arm-none-eabi-gcc` toolchain can also generate big endian code and the toolchain is used by the build system only to compile C code from source -- it doesn't link to binaries provides by the toolchain (like startup object or libc.a).

Furthermore this toolchain doesn't contain pre-compiled binaries for the v7-R architecture (or the M profile):

``` console
$ arm-none-eabi-readelf -A $(find -name crti.o) | grep _profile
  Tag_CPU_arch_profile: Application
  Tag_CPU_arch_profile: Application
  Tag_CPU_arch_profile: Application
  Tag_CPU_arch_profile: Application
  Tag_CPU_arch_profile: Application
(this repeats 20+ more times)
```